pages/learn/develop/dockerfile.md

```diff
@@ -101,6 +101,18 @@ Here are the supported machine names and architectures:
 
 {{> "general/deviceTypeNames"}}
 
+## Multiple Dockerfiles
+
+There are cases where just specifying a Dockerfile template doesn't work, as different systems or architectures may need different configuration and installation files. The solution to this problem is having multiple dockerfiles that are automatically selected based on the device it is going to be used.
+
+When using the command `balena push`, the balena CLI tool looks at all available dockerfiles and builds the appropriate container based on the following order:
+
+* Dockerfile.<device-type>
+* Dockerfile.<arch>
+* Dockerfile.template
+
+As an example, if you have two dockerfiles in the directory, `Dockerfile.raspberrypi3` and `Dockerfile.template`, whenever you run `balena push <app-name>` if the `device-type` is a Raspberry Pi 3, it will select `Dockerfile.raspberrypi3`  and for all other devices it will automatically select `Dockerfile.template`.
```

Comment:
Perhaps better not to focus on `balena push` only, as the feature is available through `git push`, `balena push`, `balena build` and `balena deploy`.